en/orm/retrieving-data-and-resultsets.rst

```diff
@@ -497,8 +497,9 @@ load fields off of contained associations, you can use ``autoFields()``::
 
     // Select id & title from articles, but all fields off of Users.
     $query->select(['id', 'title'])
-        ->contain(['Users'])
-        ->autoFields(true);
+        ->contain(['Users' => function($q) {
+			return $q->autoFields(true);
+		          }]);
```

Comment:
>  You mean that since hasOne and belongsTo built with the original query, autoFields() should not have any effects on the original, is that it ?

Yes, because [of how attachTo()](https://github.com/cakephp/cakephp/blob/master/src/ORM/Association.php#L531-L543) uses a dummy query, and then [appends the fields](https://github.com/cakephp/cakephp/blob/master/src/ORM/Association.php#L676-L699) to the original query.

However, if the code is working different in practice I am probably misreading the code. In either case we can merge these changes in.